gn2/wqflask/oauth2/session.py

```diff
@@ -118,3 +119,10 @@ def toggle_token_refreshing():
 def is_token_refreshing():
     """Returns whether the token is being refreshed or not."""
     return session_info().get("token_refreshing", False)
+
+def set_redirect_url(url):
```

Comment:
This looks good, but you should probably add a related `clear_redirect_url` function to clean up after yourself once the login is complete, but just before you do the redirect.